xdsl/parser.py

```diff
@@ -1646,7 +1646,53 @@ def _get_op_by_name(self, span: Span) -> type[Operation]:
 
         self.raise_error(f"Unknown operation {op_name}!", span)
 
-    def parse_optional_region(self) -> Region | None:
+    @dataclass
+    class Argument:
+        """
+        A block argument parsed from the assembly.
+        Arguments should be parsed by `parse_argument` or `parse_optional_argument`.
+        """
+
+        name: Span
+        """The name as displayed in the assembly."""
+
+        type: Attribute | None
+        """The type of the argument, if any."""
```

Comment:
Why is this a member of the parent class? Shouldn't this be a standalone thing?

Reply:
I made this as a member of the Parser since we shouldn't really use this outside of it.
It's the same with the `Delimiter` class. I don't really like having an `Argument` toplevel class, especially since the user is not supposed to use it. Also, I don't want to make it private, as users might still want to add a type annotation for it.